File: Software/Robot_Path/Robot_Path_Planning/map.cpp
#include "map.h"

Map::Map(float width, float length):
  width_(width),
  length_(length)
{
}

list<Robot> Map::list_robot() const
{
    return list_robot_;
}

void Map::setList_robot(const list<Robot> &list_robot)
{
    list_robot_ = list_robot;
}

list<Obstacles> Map::list_obstacles() const
{
    return list_obstacles_;
}

void Map::setList_obstacles(const list<Obstacles> &list_obstacles)
{
    list_obstacles_ = list_obstacles;
}

float Map::width_resolution() const
{
  return width_resolution_;
}

void Map::setWidth_resolution(float width_resolution)
{
  width_resolution_ = width_resolution;
}

float Map::length_resolution() const
{
  return length_resolution_;
}

void Map::setLength_resolution(float length_resolution)
{
  length_resolution_ = length_resolution;
}

File: cdr_robot_ws/src/software/src/robot_path/Map.py
#!usr/bin/env python-

import numpy as np
import numpy.linalg as alg

import Obstacle from Obstacle
import Robot from Robot

class Map:
    def __init__(self, width, length, width_resolution, length_resolution):
        self.width_ = width
        self.length_ = length
        self.width_resolution_ = width_resolution
        self.length_resolution_ = length_resolution
        self.matrix_= np.zeros((width_resolution,length_resolution)) #Matrix set to 0 
        self.obstacles_= set()
        self.robots_ = set()

    def Add_Robot(self, robot):
        self.matrix_[robot.pose2D] = 2 #Robot is on the map and set to 2

    def Add_Obstacle(self, obstacle):
        self.matrix_[obstacle.pose2D] = 1 #Obstacle i son the map and set to 1

    def Pose_Available(self, piece):
        if(self.matrix_[piece.pose2D] == 0):
            return True
        else:
            return False